pkg/pullrequest/api.go

```diff
@@ -246,34 +255,41 @@ func (h *Handler) createNewComments(ctx context.Context, comments []*scm.Comment
 			Body: dc.Body,
 		}
 		h.logger.Infof("Creating comment %s for PR %d", c.Body, h.prNum)
-		if _, _, err := h.client.Issues.CreateComment(ctx, h.repo, h.prNum, c); err != nil {
-			h.logger.Warnf("Error creating comment: %v", err)
-			merr = multierror.Append(merr, err)
+		if _, _, err := h.client.PullRequests.CreateComment(ctx, h.repo, h.prNum, c); err != nil {
+			merr = multierror.Append(merr, xerrors.Errorf("creating comment: %s", c, err))
 		}
 	}
 	return merr
 }
 
-func (h *Handler) uploadStatuses(ctx context.Context, status *scm.CombinedStatus) error {
-	if status == nil {
+func (h *Handler) uploadStatuses(ctx context.Context, statuses []*scm.Status) error {
+	if statuses == nil {
+		h.logger.Info("Skipping statuses, nothing to set.")
 		return nil
 	}
 
-	cs, _, err := h.client.Repositories.FindCombinedStatus(ctx, h.repo, status.Sha)
+	// We have to retrieve the PR to set the status one more time, the commits could have changed.
+	pr, _, err := h.client.PullRequests.Find(ctx, h.repo, h.prNum)
```

Comment:
We're doing this just to get the SHA, right? We should use the value from the local PR, since this value may have changed during pipeline execution. Easiest thing to do would be to pass it in as a param from the [PullRequest](https://godoc.org/github.com/jenkins-x/go-scm/scm#PullRequest) struct.

Reply:
Nice catch!